libpod/container_inspect.go

```diff
@@ -211,7 +211,11 @@ func (c *Container) getContainerInspectData(size bool, driverData *define.Driver
 		return nil, err
 	}
 	data.NetworkSettings = networkConfig
-	addInspectPortsExpose(c.config.ExposedPorts, data.NetworkSettings.Ports)
+	// Ports in NetworkSettings includes exposed ports for network modes that are not host,
+	// and not container.
+	if !(c.config.NetNsCtr !x= "" || c.NetworkMode() == "host") {
```

Comment:
typo

but personally I would have a slight preference for `if c.config.NetNsCtr == "" && c.NetworkMode() != "host"` as I think this is clearer but it should be logically the same